automerge-backend/src/object_store.rs

```diff
@@ -279,6 +342,27 @@ impl ObjectStore {
         }
     }
 
+    pub fn generate_diffs(&self) -> Vec<Diff> {
+        let mut diffs = vec![];
+        let mut seen = HashSet::new();
+        let mut next : Vec<ObjectID> = vec![ ObjectID::Root ];
+
+        while !next.is_empty() {
```

Comment:
I think this would probably be nicely expressed as `while let Some(next) = next {...}`